src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerControllerTest.java

```diff
@@ -53,30 +60,61 @@ public void given_eventInformation_from_ccd_should_respond_with_200() {
             .body("size()", is(1))
             .body("[0].name", is("Process Application"))
             .body("[0].formKey", is("processApplication"))
+            .assertThat().body("[0].id", notNullValue())
             .extract()
             .path("[0].id");
 
     }
 
-    @After
-    public void cleanUpTask() {
-        System.out.println("*** taskId: " + taskId);
-        given()
-            .header(SERVICE_AUTHORIZATION, s2sToken)
-            .accept(APPLICATION_JSON_VALUE)
-            .contentType(APPLICATION_JSON_VALUE)
-            .when()
-            .post(camundaUrl + "/task/{task-id}/complete", taskId);
+    @Test
+    public void given_eventInformation_from_ccd_should_cancel_task_and_respond_with_204() {
+        String eventToCancelTask = "submitReasonsForAppeal";
+        String previousStateToCancelTask = "awaitingReasonsForAppeal";
+
+        EventInformation eventInformation = EventInformation.builder()
+            .eventInstanceId("some event instance Id")
+            .dateTime(LocalDateTime.now().plusDays(2))
+            .caseReference(caseId)
+            .jurisdictionId("IA")
+            .caseTypeId("Asylum")
+            .eventId(eventToCancelTask)
+            .newStateId("")
+            .previousStateId(previousStateToCancelTask)
+            .userId("some user Id")
+            .build();
 
         given()
             .contentType(APPLICATION_JSON_VALUE)
-            .accept(APPLICATION_JSON_VALUE)
-            .header(SERVICE_AUTHORIZATION, s2sToken)
-            .baseUri(camundaUrl)
+            .body(asJsonString(eventInformation))
             .when()
-            .get("/history/task?taskId=" + taskId)
+            .post("/messages")
             .then()
-            .body("[0].deleteReason", is("completed"));
+            .statusCode(HttpStatus.NO_CONTENT.value());
+
+    }
+
+    @After
+    public void cleanUpTask() {
```

Comment:
maybe not now but so we don't repeat ourselves we could add this to the SpringBootFunctionalBaseTest or a common class

Reply:
Noted. I'll keep it as it for now but definitely will refactor when the need arises. Thanks.